pyserver/main.py

```diff
@@ -0,0 +1,362 @@
+ 
+  # count the claims in each subtopic 
+  for claim in comms_to_claims:
+    if "topicName" in claim:
+      if claim["topicName"] in node_counts:
+        node_counts[claim["topicName"]]["total"] += 1
+        if "subtopicName" in claim:
+          if claim["subtopicName"] in node_counts[claim["topicName"]]["subtopics"]:
+            node_counts[claim["topicName"]]["subtopics"][claim["subtopicName"]]["total"] += 1
+            node_counts[claim["topicName"]]["subtopics"][claim["subtopicName"]]["claims"].append(claim["claim"])
+          else:
+            node_counts[claim["topicName"]]["subtopics"][claim["subtopicName"]] = { "total" : 1, "claims" : [claim["claim"]]}
+      else:
+        node_counts[claim["topicName"]] = {"total" : 1, "subtopics" : {claim["subtopicName"] : {"total" : 1, "claims" : [claim["claim"]]}}}
+
+
+  if log_to_wandb:
+    wandb.init(project = config.WANDB_PROJECT_NAME,
+               config={"model" : config.MODEL})
+    wandb.log({
+      "u/2/N_tok" : TK_2_TOT,
+      "u/2/in_tok": TK_2_IN,
+      "u/2/out_tok" : TK_2_OUT,
+      "rows_to_claims" : wandb.Table(
+                           data=comms_to_claims_html,
+                           columns = ["comments", "claims"])
+    })
+ 
+  net_usage = {"total_tokens" : TK_2_TOT,
+               "prompt_tokens" : TK_2_IN,
+               "completion_tokens" : TK_2_OUT}
+
+
+  return {"claims_tree" : node_counts, "usage" : net_usage}
+
+def dedup_claims(claims:list)-> dict:
+  """
+  Given a list of claims for a given subtopic, identify which ones are near-duplicates
+  """
+  client = OpenAI()
+
+  # add claims with enumerated ids (relative to this subtopic only)
+  full_prompt = config.CLAIM_DEDUP_PROMPT
+  for i, orig_claim in enumerate(claims):
+    full_prompt += "\nclaimId"+str(i)+ ": " + orig_claim
+
+  response = client.chat.completions.create(
+    model = config.MODEL,
+    messages = [
+      {
+        "role": "system",
+        "content": config.SYSTEM_PROMPT
+      },
+      {
+            "role": "user",
+            "content": full_prompt
+      }
+    ],
+    temperature = 0.0,
+    response_format = {"type": "json_object"}
+  )
+  deduped_claims = response.choices[0].message.content
+  return {"dedup_claims" : json.loads(deduped_claims), "usage" : response.usage}
+
+#####################################
+# Step 3: Sort & deduplicate claims #
+#-----------------------------------#
+@app.put("/sort_claims_tree/")
+def sort_claims_tree(claims_tree:ClaimTree, sort_type : str = "deduplicate", log_to_wandb: bool = False)-> dict:
+  #TODO: do we ever want to deduplicate without sorting? or sort without deduplicating?
```

Comment:
probably not.

eventually we'll want to sort by number of people who mentioned a topic instead of number of claims, but probably still fine to keep that in the same function